src/Avalonia.Animation/DisposeAnimationInstanceObservable.cs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) The Avalonia Project. All rights reserved.
+// Licensed under the MIT license. See licence.md file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reactive.Linq;
+using Avalonia.Animation.Utils;
+using Avalonia.Collections;
+using Avalonia.Data;
+using Avalonia.Reactive;
+
+namespace Avalonia.Animation
+{
+    /// <summary>
+    /// Manages the lifetime of animation instances as determined by its selector state.
+    /// </summary>
+    internal class DisposeAnimationInstanceObservable<T> : IObserver<bool>, IDisposable
```

Comment:
This is called an "Observable" but it only implements `IObserver` and in addition, the variable it's assigned to is called `subject`. Probably should decide what it is ;)

Reply:
Oof, sorry. Forgot to rename that class name to Subject

Reply:
Hmm but it's not a subject because a subject implements both `IObserver` and `IObservable`. Sorry, I realise this is getting a bit nit-picky!